### PR TITLE
Use incremental get for batch_get_command

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

###  What have you changed?

- use incremental get of `SnapStore` to perform batch get.

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test

###  Benchmark result if necessary (optional)

2~3% improvement in oltp_point_select.

### CODE REVIEW

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

File: src/storage/mod.rs [hunk 1]
```diff
@@ -990,15 +992,34 @@ impl<E: Engine, L: LockMgr> Storage<E, L> {
                                 !ctx.get_not_fill_cache(),
                             );
                             let mut results = vec![];
-                            // TODO: optimize using seek.
+                            let use_incremental_get = gets.len() >= BATCH_INCREMENTAL_GET_LIMIT;
+                            if use_incremental_get {
+                                gets.sort_by(|a, b| match a.key.cmp(&b.key) {
```
Comment (lonng) [inline src/storage/mod.rs]: Do you want to sort by `key` desc? How about `gets.sort_by(|a, b| match b.key.cmp(&a.key)`

Comment (lonng) [inline src/storage/mod.rs]: How about
```rust
gets.sort_by(|a, b| match a.key.cmp(&b.key) {
    cmp::Ordering::Equal => b.ts.cmp(a.ts),
    ord => ord
}
```

Comment (tabokie) [reply src/storage/mod.rs]: nice!

File: src/storage/mod.rs [hunk 2]
```diff
@@ -990,15 +992,30 @@ impl<E: Engine, L: LockMgr> Storage<E, L> {
...
                                 snap_store.set_start_ts(get.ts.unwrap());
                                 snap_store.set_isolation_level(get.ctx.get_isolation_level());
-                                results.push(
-                                    snap_store
-                                        .get(&get.key, &mut statistics)
-                                        .map_err(Error::from),
-                                );
+                                if use_incremental_get {
+                                    results.push(
+                                        snap_store.incremental_get(&get.key).map_err(Error::from),
```
Comment (breeswish) [inline src/storage/mod.rs]: Do we need to restore the order?

File: src/storage/mod.rs [hunk 3]
```diff
@@ -990,15 +992,30 @@ impl<E: Engine, L: LockMgr> Storage<E, L> {
                                 !ctx.get_not_fill_cache(),
                             );
                             let mut results = vec![];
-                            // TODO: optimize using seek.
+                            let use_incremental_get = gets.len() >= BATCH_INCREMENTAL_GET_LIMIT;
+                            if use_incremental_get {
```
Comment (breeswish) [inline src/storage/mod.rs]: Incremental get can be always used (it supports unordered get now). Sort can be optionally performed.

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by mahjonp** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 512
```

**Comment by tabokie** [comment]
PTAL @zhangjinpeng1987 @Little-Wallace

**Comment by tabokie** [comment]
/bench
```
workload:
  threads: 1024
```

**Pull request closed without merging**